examples/platforms/utils/soft_source_match_table.h

```diff
@@ -53,16 +53,25 @@ extern "C" {
 #define RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM OPENTHREAD_CONFIG_MLE_MAX_CHILDREN
 #endif
 
+#ifndef RADIO_CONFIG_SRC_MATCH_PANID_NUM
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+#define RADIO_CONFIG_SRC_MATCH_PANID_NUM 3
+#else // !OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+#define RADIO_CONFIG_SRC_MATCH_PANID_NUM 1
+#endif // OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+#endif // RADIO_CONFIG_SRC_MATCH_PANID_NUM
+
 #if RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM || RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM
-void utilsSoftSrcMatchSetPanId(uint16_t aPanId);
+void    utilsSoftSrcMatchSetPanId(uint8_t iid, uint16_t aPanId);
+uint8_t utilsSoftSrcMatchFindIidFromPanId(otPanId panId);
 #endif // RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM || RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM
 
 #if RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM
-int16_t utilsSoftSrcMatchShortFindEntry(uint16_t aShortAddress);
+int16_t utilsSoftSrcMatchShortFindEntry(uint8_t iid, uint16_t aShortAddress);
 #endif // RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM
 
 #if RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM
-int16_t utilsSoftSrcMatchExtFindEntry(const otExtAddress *aExtAddress);
+int16_t utilsSoftSrcMatchExtFindEntry(uint8_t iid, const otExtAddress *aExtAddress);
```

Comment:
Agree that it is more general utility APIs. But we would need iid provision to source match table to iterate over the entries per network.  Also there might have same checksum for two entries when there are multiple networks.

I think we will maintain the copy of this internally so that we don't modify these APIs.

Reply:
I would suggest option 2.

But even more general, I think would be better to have each protocol/network decide and implement its own source match table (and not reuse the implementation from OT lib).

Reply:
Sounds good. Reverted the changes from source match table to maintain the separate copy.